### PR TITLE
fix: Make sure that cache is locked when updating for contextifyFramesIntegration

### PR DESCRIPTION
**Pull request by kamilogorek**
Fixes https://github.com/getsentry/sentry-go/issues/55

### CODE REVIEW

File: integrations.go [hunk 1]
```diff
@@ -374,3 +376,9 @@ func (cfi *contextifyFramesIntegration) addContextLinesToFrame(frame Frame, line
 	}
 	return frame
 }
+
+func (cfi *contextifyFramesIntegration) updateCachedLocation(key, val string) {
+	cfi.mu.Lock()
```
Comment (bruno-garcia) [inline integrations.go]: This is a write-lock. I can't spot a use of a read-lock. Is the reader-write really needed here as opposed to a plain mutex?

File: integrations.go [hunk 2]
```diff
@@ -352,12 +354,12 @@ func (cfi *contextifyFramesIntegration) findNearbySourceCodeLocation(originalPat
 		possibleLocation := strings.Join(components, "/")
 
 		if fileExists(possibleLocation) {
-			cfi.cachedLocations[originalPath] = possibleLocation
+			cfi.updateCachedLocation(originalPath, possibleLocation)
```
Comment (bruno-garcia) [inline integrations.go]: You need a `cfi.mu.RLock()` here.
If this collection isn't thread-safe, we must make sure no read happens while we're writing into it. If I understand correctly you are guarding the writes to it via `cfi.mu.Lock()` but not the reads with `cfi.mu.RLock()`.

File: integrations_test.go [hunk 3]
```diff
@@ -224,3 +225,15 @@ func TestContextifyFramesNonexistingFilesShouldNotDropFrames(t *testing.T) {
 	contextifiedFrames := cfi.contextify(frames)
 	assertEqual(t, len(contextifiedFrames), len(frames))
 }
+
+func TestContextifyFramesConcurrentCacheUpdates(t *testing.T) {
+	cfi := contextifyFramesIntegration{
```
Comment (bruno-garcia) [inline integrations_test.go]: Does this test repro the issue before the fix is applied?

Comment (kamilogorek) [reply integrations_test.go]: Yes. I made sure it fails first. Although I wasnt able to do this for reads, thus I didnt change them anywhere.

**Pull request merged**